src/include/daos_obj.h

```diff
@@ -1129,13 +1129,97 @@ daos_oit_close(daos_handle_t oh, daos_event_t *ev);
  * \return		These values will be returned by \a ev::ev_error in
  *			non-blocking mode:
  *			0		Success
- *			-DER_NO_HDL	Invalid object open handle
+ *			-DER_NO_HDL	Invalid OIT open handle
  *			-DER_INVAL	Invalid parameter
  */
 int
 daos_oit_list(daos_handle_t oh, daos_obj_id_t *oids, uint32_t *oids_nr,
 	      daos_anchor_t *anchor, daos_event_t *ev);
 
+#define DAOS_OIT_MARKER_MAX_LEN	(32)
+
+/**
+ * Mark an object ID in the Object Index Table (OIT).
+ *
+ * \param[in]	oh	OIT open handle.
+ * \param[in]	oid	object ID in the OIT.
+ * \param[in]	marker	the data/status to be marked for the OID, the max valid length (in bytes)
+ *			is DAOS_OIT_MARKER_MAX_LEN. NULL is invalid parameter.
+ * \param[in]	ev	Completion event, it is optional and can be NULL.
+ *			Function will run in blocking mode if \a ev is NULL.
+ *
+ * \return		These values will be returned by \a ev::ev_error in
+ *			non-blocking mode:
+ *			0		Success
+ *			-DER_NO_HDL	Invalid OIT open handle
+ *			-DER_INVAL	Invalid parameter
+ *			-DER_NONEXIST	Input OID not in the OIT.
+ */
+int
+daos_oit_mark(daos_handle_t oh, daos_obj_id_t oid, d_iov_t *marker, daos_event_t *ev);
```

Comment:
if this is for single OID, then probably should be daos_oid_mark() ? Again should we add epoch here?

Reply:
keep "daos_oit_" prefix can means it is only for OIT table, not for generic object/oid.
"should we add epoch here" why need to add epoch? you mean as the update epoch?it probably fine to use latest hlc for update, and also for list()?